### PR TITLE
Utiliser htmlproofer ainsi que html5validator

### PR DESCRIPTION
**Pull request by maukoquiroga**
L'idée étant de forcer un check up plus complet du site ~~(e.g: https).~~

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+html5validator
```
Comment (MattiSG) [inline requirements.txt]: I don't think using inference is worth one more file at the root.

File: circle.yml [hunk 2]
```diff
@@ -1,15 +1,16 @@
...
-
+    - bundle exec jekyll --version # print it out
 test:
   override:
     - bundle exec jekyll doctor
     - bundle exec jekyll build
-    - grep "css/main.css" _site/index.html  # smoke test that something was actually compiled
+    - bundle exec htmlproofer ./_site --assume-extension --check-html --disable-external --empty-alt-ignore --enforce-https --url-ignore "/labonneboite|modernisation/"
     - html5validator --root _site
+    - grep "css/main.css" _site/index.html # smoke test that something was actually compiled
```
Comment (MattiSG) [inline circle.yml]: How would htmlproofer handle an empty directory? If it shouts, this smoke test is not useful anymore.

Comment (MattiSG) [inline circle.yml]: > Smoke testing an output was created makes more sense before trying to lint it.

File: circle.yml [hunk 3]
```diff
@@ -1,15 +1,16 @@
...
-    - bundle exec jekyll --version  # print it out
-    - sudo pip install html5validator
-
+    - bundle exec jekyll --version # print it out
 test:
   override:
     - bundle exec jekyll doctor
     - bundle exec jekyll build
-    - grep "css/main.css" _site/index.html  # smoke test that something was actually compiled
+    - bundle exec htmlproofer ./_site --assume-extension --check-html --disable-external --empty-alt-ignore --enforce-https --url-ignore "/labonneboite|modernisation/"
```
Comment (MattiSG) [inline circle.yml]: Mmh. I'm not sure this `enforce-https` + whitelist scales.

Ok to try, but we need to document how to handle such validation errors (i.e. “update the whitelist”) in the CONTRIBUTING.

**Comment by MattiSG** [comment]
Which checks `html5validator` does that Nokogiri's validator does not?

File: circle.yml (hunk 3 above)
Comment (maukoquiroga) [inline circle.yml]: I was actually thinking it won't scale.

File: circle.yml (hunk 2 above)
Comment (MattiSG) [inline circle.yml]: Hahaha just discovered that `html5validator` _already_ [shouted](https://circleci.com/gh/sgmap/beta.gouv.fr/451)  :wink:

File: _includes/head.html [hunk 4]
```diff
@@ -6,13 +6,13 @@
...
 	<meta name="description" content="{% if page.excerpt %}{{ page.excerpt | strip_html | strip_newlines | truncate: 160 }}{% else %}{{ site.description }}{% endif %}">
 
-	<link rel="stylesheet" type="text/css" class="ui" href="//oss.maxcdn.com/semantic-ui/2.1.8/components/reset.min.css">
-	<link rel="stylesheet" type="text/css" class="ui" href="//oss.maxcdn.com/semantic-ui/2.1.8/components/container.min.css">
-	<link rel="stylesheet" type="text/css" class="ui" href="//oss.maxcdn.com/semantic-ui/2.1.8/components/grid.min.css">
-	<link rel="stylesheet" type="text/css" class="ui" href="//oss.maxcdn.com/semantic-ui/2.1.8/components/divider.min.css">
-	<link rel="stylesheet" type="text/css" class="ui" href="//oss.maxcdn.com/semantic-ui/2.1.8/components/card.min.css">
-	<link rel="stylesheet" type="text/css" class="ui" href="//oss.maxcdn.com/semantic-ui/2.1.8/components/image.min.css">
-	<link rel="stylesheet" type="text/css" class="ui" href="//oss.maxcdn.com/semantic-ui/2.1.8/components/transition.min.css">
+	<link rel="stylesheet" type="text/css" class="ui" href="https://oss.maxcdn.com/semantic-ui/2.1.8/components/reset.min.css">
```
Comment (MattiSG) [inline _includes/head.html]: :+1:

**Comment by MattiSG** [comment]
> Which checks html5validator does that Nokogiri's validator does not?

Ok, found at least one: unescaped spaces in URLs are detected by `html5validator`, not by `html-proofer`. Let's keep both for now.

**Comment by MattiSG** [comment]
Very strange: if I execute `htmlproofer` with the same options locally, I get the following errors:

```
- ./_site/contact.html
  *  internally linking to /css/text.css, which does not exist (line 24)
- ./_site/philosophie.html
  *  internally linking to /css/text.css, which does not exist (line 24)
```

…while CI says everything is fine.

**Comment by maukoquiroga** [comment]
Could you give me the results of the following commands?

```sh
ruby -v
gem list | grep htmlproofer
echo $GEM_PATH
bundle show html-proofer
```

**Comment by MattiSG** [comment]
```
› ruby -v
ruby 2.1.7p400 (2015-08-18 revision 51632) [x86_64-darwin15.0]
› gem list | grep htmlproofer
# $? = 1 
› echo $GEM_PATH

› bundle show html-proofer
/Users/sgmap/.rbenv/versions/2.1.7/lib/ruby/gems/2.1.0/gems/html-proofer-3.0.5
```

File: circle.yml [hunk 5]
```diff
@@ -1,4 +1,6 @@
 machine:
+  python:
+    version: 3.5.1
```
Comment (MattiSG) [inline circle.yml]: Nope. CI build goes from 1'30 to 4'30 just to download and set up Python & friends.

**Comment by maukoquiroga** [comment]
fixup + rebase

**Pull request merged**